pkg/core/service.go

```diff
@@ -86,6 +93,105 @@ func newService(options CreateServiceOptions) v1alpha1.Service {
 			},
 		},
 	}
+
+	if len(options.Env) > 0 {
+		var envVars []core_v1.EnvVar
+		for _, env := range options.Env {
+			envEntry := strings.SplitN(env, "=", 2)
+			if len(envEntry) != 2 {
+				return v1alpha1.Service{}, errors.New(fmt.Sprintf("unable to parse '%v', environment variables must provided as 'name=value'", env))
+			}
+			envKey := strings.TrimSpace(envEntry[0])
```

Comment:
I'm not keen on trimming whitespace here as it makes the documentation of the syntax (when we write it) more fussy than necessary. I don't really see why users can't learn to get the strict syntax correct from day 1.